packages/council-graphql/src/resolvers/index.ts

```diff
@@ -59,40 +64,67 @@ export const resolvers: Resolvers<CouncilContext> = {
         voters = await VoterModel.getAll({ context });
       }
 
-      if (search) {
-        const candidates: string[] = [];
-
-        voters = await Promise.all(
-          voters.map(async (voter) => {
-            if (!voter) {
-              return;
-            }
+      if (search || votingPowerMin || votingPowerMax) {
+        let vaults: (VotingVault | undefined)[];
 
-            candidates.push(voter.address);
+        for (let voter of voters) {
+          if (!voter) {
+            continue;
+          }
 
+          // Get ensName of voter
+          if (search) {
             const ensName = await VoterModel.getEnsName({ voter, context });
-            if (ensName) {
-              candidates.push(ensName);
+            voter = { ...voter, ensName };
+          }
+
+          // Get votingPower of voter
+          if (votingPowerMin || votingPowerMax) {
+            if (votingVaults) {
+              vaults = VotingVaultModel.getByAddresses({
+                addresses: votingVaults,
+                context,
+              }).filter((vault) => vault !== undefined);
+            } else {
+              vaults = VotingVaultModel.getAll({ context });
             }
 
-            return {
-              ...voter,
-              ensName,
-            };
-          }),
-        );
+            const votingPower = await VotingPowerModel.getByVoter({
+              voter,
+              votingVaults: vaults as VotingVault[],
+              context,
+            });
+            voter = { ...voter, votingPower };
+          }
 
-        const matches = filter(candidates, search);
+          let searchCondition = true;
+          let votingPowerCondition = true;
+          if (search) {
+            // Check whether current voter passes search filter
+            searchCondition = !!filter(
+              [voter.ensName || "", voter.address],
+              search,
+            ).length;
```

Comment:
I know we've removed a loop here, but I think it may still be more efficient to only run `filter` once on the whole list rather than on each voter. I don't think it needs to be refactored again right now, but fuzzy search is likely more expensive than a loop over the voters, so you probably want to limit the number of calls to it when possible.